textrepo-app/src/main/java/nl/knaw/huc/service/task/deleter/JdbiDeleteDocumentTaskBuilder.java

```diff
@@ -23,22 +28,40 @@ public DeleteDocumentTaskBuilder forExternalId(String externalId) {
     return this;
   }
 
+  @Override
+  public DeleteDocumentTaskBuilder withIndexing(boolean indexing) {
+    this.indexing = indexing;
+    return this;
+  }
+
   @Override
   public Task<Document> build() {
-    return new DeleteDocumentTask(externalId);
+    return new DeleteDocumentTask(externalId, indexing, indexService);
   }
 
   private class DeleteDocumentTask implements Task<Document> {
     private final String externalId;
+    private final boolean indexing;
+    private final IndexService indexService;
 
-    private DeleteDocumentTask(String externalId) {
+    private DeleteDocumentTask(String externalId, boolean indexing, IndexService indexService) {
       this.externalId = externalId;
+      this.indexing = indexing;
+      this.indexService = indexService;
     }
 
     @Override
     public Document run() {
       return jdbi.inTransaction(transaction -> {
         final var doc = new FindDocumentByExternalId(externalId).executeIn(transaction);
+
+        if (indexing) {
+          transaction
+              .attach(DocumentFilesDao.class)
+              .findFilesByDocumentId(doc.getId())
+              .forEach(file -> indexService.delete(file.getId()));
```

Comment:
Encapsuleren in een RemoveFromIndex() object?

Reply:
Zie 481ce921